### ISSUE TITLE
Agent tracked_tensors method doesn't seem to work correctly

### ISSUE THREAD

**Issue by Delamea** [tensorforce/tensorforce#817]
Hi everyone,

I have encountered a difficulty in my use of tensorforce and after some research I still can't identify the cause. I'm looking for a way to get the return over episodes for a training with the runner utility. I added the tracking parameter to the definition of my agent with the value "all" but when I call the tracked_tensors() method it returns very little information and I can't determine the cause.

Here is an example of code that does not return the result I want.

`def main():
    # OpenAI-Gym environment specification
    environment = dict(environment='gym', level='CartPole-v1')
    # or: environment = Environment.create(
    #         environment='gym', level='CartPole-v1', max_episode_timesteps=500)

    # PPO agent specification
    agent = dict(
        agent='ppo',
        # Automatically configured network
        network='auto',
        # PPO optimization parameters
        batch_size=10, update_frequency=2, learning_rate=3e-4, multi_step=10,
        subsampling_fraction=0.33,
        # Reward estimation
        likelihood_ratio_clipping=0.2, discount=0.99, predict_terminal_values=False,
        # Baseline network and optimizer
        baseline=dict(type='auto', size=32, depth=1),
        baseline_optimizer=dict(optimizer='adam', learning_rate=1e-3, multi_step=10),
        # Regularization
        l2_regularization=0.0, entropy_regularization=0.0,
        # Preprocessing
        state_preprocessing='linear_normalization', reward_preprocessing=None,
        # Exploration
        exploration=0.0, variable_noise=0.0,
        # Default additional config values
        config=None,
        # Save agent every 10 updates and keep the 5 most recent checkpoints
        saver=dict(directory='model', frequency=10, max_checkpoints=5),
        # Log all available Tensorboard summaries
        summarizer=dict(directory='summaries', summaries='all'),
        # Do not record agent-environment interaction trace
        recorder=None,
        tracking='all'
    )
    # or: Agent.create(agent='ppo', environment=environment, ...)
    # with additional argument "environment" and, if applicable, "parallel_interactions"

    # Initialize the runner
    runner = Runner(agent=agent, environment=environment, max_episode_timesteps=500)

    # Train for 200 episodes
    runner.run(num_episodes=20)
    print(runner.agent.tracked_tensors())
    runner.close()

    # plus agent.close() and environment.close() if created separately

if __name__ == '__main__':
    main()`

I obtain the following result:
`{'agent/policy/action_distribution/probabilities': array([0.3002068, 0.6997932], dtype=float32)}`

Why does the dictionary only contain this?

Thank you for your help!

**Comment by AlexKuhnle** [comment]
Yes, you need to build your own utility which stores the history, which should be easily doable by augmenting the act-observe loop, and `tracked_tensors()` is a good way to get access to some of the internal values, like losses etc.

**Comment by Delamea** [comment]
Hi @AlexKuhnle 

As discussed above I have implemented a custom runner utility. However I still have difficulties with the `agent.tracked_tensors()` method. To build the history I call it if there is an update following the code you gave above.

However I have the same result as in my first post with only the values of the distribution parameters.

I explored a little the code of the library and I was surprised to see that if I try to print the tracked tensors after an update using `print(agent.model.tracking_variables)` the dictionary was empty. Maybe there is a problem here...

I'm using Tensorforce 0.6.4.

Thank for you help!
